src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -134,12 +139,16 @@ public void setPeriod(String period) {
     public void visitJavadocToken(DetailNode ast) {
         String firstSentence = getFirstSentence(ast);
         final int endOfSentence = firstSentence.lastIndexOf(period);
-        if (endOfSentence == -1) {
-            if (!isOnlyInheritDoc(ast)) {
-                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
-            }
+        final String summaryDoc = getSummarySentence(ast);
```

Comment:
@sagar-shah94 , please explain why we have two method `getFirstSentence` and `getSummarySentence` ?
This methods should do absolutely the same as first sentence is summary sentence, please use only one.
If smth is missed in one of them, please merge them.